### PR TITLE
A list of low hanging docs changes from our feature testing

### PR DESCRIPTION
**Pull request by csibbitt**
Items below are from STF-1167

* OSP Connection 1a Docs: Check other containers, not just metrics_qdr
* Dashboards 1c: Docs: Rewording or change object we source creds from
* Dashboards 1d: Docs: Wait for grafana restart
* Metrics retention 1b:  Docs: Tell customers how to verify
* Alerts 2: Docs: More explicit example of how to construct this config
* HA 2 Docs: Improve alertmanager config docs and re-test
  * Continues to work for me
* Ephemeral Storage Docs: Document this only upstream and/or dev docs
* Observability Strategy Docs: Add a note of which objects to delete
* Docs: Double check that libpodstats and sensubility are not mentioned in OSP 13 (they shouldn't be)
  * I checked here: https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/13/html-single/service_telemetry_framework_1.5/index
  * libpodstats is not mentioned, but sensubility is mentioned in several places
  * Is this a mistake? I don't have an OSP 13 handy and haven't dug through the artifacts to figure it out.
    * If changes are needed, there are several affected locations, so I'll use a dedicated PR
* Certificate Handling (Issue 15.1 or sooner): Set 7.5yr expiry on all certs
  * https://github.com/infrawatch/service-telemetry-operator/pull/389

### CODE REVIEW

**Comment by csibbitt** [comment]
Just a draft for now, I haven't even rendered it; but I wanted to get through as many of the easy docs changes as possible in one go.

File: doc-Service-Telemetry-Framework/modules/proc_creating-an-alert-route-with-templating-in-alertmanager.adoc [hunk 1]
```diff
@@ -36,40 +36,41 @@ If the `alertmanagerConfigManifest` parameter contains a custom template, for ex
...
 ----
 
-. Edit the `ServiceTelemetry` object for your {ProjectShort} deployment:
+. Create the necesarry alermanager config in a file called alertmanager.yaml, for example:
 +
-[source,bash]
+[source,yaml]
 ----
-$ oc edit stf default
+$ cat alertmanager.yaml
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_creating-an-alert-route-with-templating-in-alertmanager.adoc:43]: I feel like you want to follow some of the patterns like `cat > alertmanager.yaml <<EOF` here.

File: doc-Service-Telemetry-Framework/modules/proc_creating-an-alert-route-with-templating-in-alertmanager.adoc [hunk 2]
```diff
@@ -36,40 +36,41 @@ If the `alertmanagerConfigManifest` parameter contains a custom template, for ex
 $ oc project service-telemetry
 ----
 
-. Edit the `ServiceTelemetry` object for your {ProjectShort} deployment:
+. Create the necesarry alermanager config in a file called alertmanager.yaml, for example:
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_creating-an-alert-route-with-templating-in-alertmanager.adoc:39]:
```suggestion
. Create the necessary alertmanager config in a file called alertmanager.yaml, for example:
```

File: doc-Service-Telemetry-Framework/modules/proc_editing-the-metrics-retention-time-period-in-service-telemetry-framework.adoc [hunk 3]
```diff
@@ -48,6 +48,19 @@ spec:
 ----
 
 . Save your changes and close the object.
+. Wait for prometheus to restart with the new settings.
++
+[source,bash]
+----
+$ oc get po -l app.kubernetes.io/name=prometheus -w
+----
+. Verify the new retention setting be checking the command line arguments used in the pod.
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_editing-the-metrics-retention-time-period-in-service-telemetry-framework.adoc:57]:
```suggestion
. Verify the new retention setting by checking the command line arguments used in the pod.
```

File: doc-Service-Telemetry-Framework/modules/proc_validating-clientside-installation.adoc [hunk 4]
```diff
@@ -14,8 +14,9 @@ TIP: Some telemetry data is available only when {OpenStackShort} has active work
 +
 [source,bash,options="nowrap",subs="attributes"]
 ----
-$ sudo {containerbin} container inspect --format '{{.State.Status}}' metrics_qdr
-
+$ sudo {containerbin} container inspect --format '{{.State.Status}}' metrics_qdr ceilometer_agent_notification collectd
```
Comment (yadneshk) [inline doc-Service-Telemetry-Framework/modules/proc_validating-clientside-installation.adoc:17]:
```suggestion
$ sudo {containerbin} container inspect --format '{{.State.Status}}' metrics_qdr ceilometer_agent_notification ceilometer_agent_central collectd
```

Comment (yadneshk) [reply doc-Service-Telemetry-Framework/modules/proc_validating-clientside-installation.adoc:17]: We could also validate `ceilometer_agent_compute` which runs only on compute nodes.

Comment (csibbitt) [reply doc-Service-Telemetry-Framework/modules/proc_validating-clientside-installation.adoc]: Okay, I added it and a note about using a different command on compute nodes.

File: doc-Service-Telemetry-Framework/modules/proc_configuring-observability-strategy.adoc [hunk 5]
```diff
@@ -23,6 +23,13 @@ spec:
 EOF
 ----
 +
+. Delete the left over objects that are managed by community operators
++
+[source,bash]
+----
+$ for o in alertmanager/default prometheus/default elasticsearch/elasticsearch grafana/default lokistack/lokistack; do oc delete $o; done
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-observability-strategy.adoc:30]: Do we need lokistack here? I know we have some loki type things, but we never really mention anything about it, so not sure how necessary that is in a standard deployment?

File: doc-Service-Telemetry-Framework/modules/proc_retrieving-and-setting-grafana-login-credentials.adoc [hunk 6]
```diff
@@ -13,11 +15,22 @@
...
 $ oc get stf default -o jsonpath="{.spec.graphing.grafana['adminUser','adminPassword']}"
 ----
 
 . To modify the default values of the Grafana administrator username and password through the ServiceTelemetry object, use the `graphing.grafana.adminUser` and `graphing.grafana.adminPassword` parameters.
++
+[source,bash]
+----
+$ oc edit stf default
+----
+. Wait for the grafana pod to restart with the new credentials in place
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_retrieving-and-setting-grafana-login-credentials.adoc:31]: I'd have to check, but I _think_ you might need a newline between the end of the source `----` line and the new bullet line. Won't get an error in the rendering, but I seem to remember the resulting rendered doc being messed up sometimes.

File: doc-Service-Telemetry-Framework/modules/proc_validating-clientside-installation.adoc [hunk 7]
```diff
@@ -10,15 +10,21 @@ TIP: Some telemetry data is available only when {OpenStackShort} has active work
...
+running
+running
+running
 running
 ----
-
++
+[NOTE]
+Use this command on compute nodes:
+sudo {containerbin} container inspect --format '{{.State.Status}}' metrics_qdr collectd ceilometer_agent_compute
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_validating-clientside-installation.adoc:26]: Missing a wrap in `[source]` or backticks.

File: doc-Service-Telemetry-Framework/modules/proc_configuring-observability-strategy.adoc (hunk 5 above)
Comment (csibbitt) [reply doc-Service-Telemetry-Framework/modules/proc_configuring-observability-strategy.adoc:30]: No? But it doesn't hurt.

We mention loki only in this one place in the docs: https://github.com/infrawatch/documentation/blob/b128d9925613749f46d91a58c8491fe10da82d13/doc-Service-Telemetry-Framework/modules/proc_creating-a-servicetelemetry-object-in-openshift.adoc#L72

I'd be just as happy to remove both references if that's what we'd prefer.

File: doc-Service-Telemetry-Framework/modules/proc_retrieving-and-setting-grafana-login-credentials.adoc (hunk 6 above)
Comment (csibbitt) [reply doc-Service-Telemetry-Framework/modules/proc_retrieving-and-setting-grafana-login-credentials.adoc:31]: image

File: doc-Service-Telemetry-Framework/modules/proc_configuring-observability-strategy.adoc (hunk 5 above)
Comment (leifmadsen) [reply doc-Service-Telemetry-Framework/modules/proc_configuring-observability-strategy.adoc:30]: Hrmmm.... I don't think I have a strong enough opinion on this to really suggest either direction over the other, so let's just go with what is here :)

I have the ticket for simply removing the loki/logging stuff out of STO slated against the STF 1.5 on OCP 4.14 epic, so I think we could adjust this at that time should we end up doing that. No change requested here.

File: doc-Service-Telemetry-Framework/modules/proc_creating-an-alert-route-with-templating-in-alertmanager.adoc [hunk 8]
```diff
@@ -36,40 +36,42 @@ If the `alertmanagerConfigManifest` parameter contains a custom template, for ex
 $ oc project service-telemetry
 ----
 
-. Edit the `ServiceTelemetry` object for your {ProjectShort} deployment:
+. Create the necessary alertmanager config in a file called alertmanager.yaml, for example:
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_creating-an-alert-route-with-templating-in-alertmanager.adoc:39]:
```suggestion
. Create the necessary Alertmanager config in a file called alertmanager.yaml, for example:
```

Comment (leifmadsen) [reply doc-Service-Telemetry-Framework/modules/proc_creating-an-alert-route-with-templating-in-alertmanager.adoc:39]: We referred to it earlier with a capital, so just working through for consistency.

File: doc-Service-Telemetry-Framework/modules/proc_creating-an-alert-route-with-templating-in-alertmanager.adoc [hunk 9]
```diff
@@ -36,40 +36,42 @@ If the `alertmanagerConfigManifest` parameter contains a custom template, for ex
...
-    data:
-      alertmanager.yaml: Z2xvYmFsOgogIHJlc29sdmVfdGltZW91dDogMTBtCiAgc2xhY2tfYXBpX3VybDogPHNsYWNrX2FwaV91cmw+CnJlY2VpdmVyczoKICAtIG5hbWU6IHNsYWNrCiAgICBzbGFja19jb25maWdzOgogICAgLSBjaGFubmVsOiAjc3RmLWFsZXJ0cwogICAgICB0aXRsZTogfC0KICAgICAgICAuLi4KICAgICAgdGV4dDogPi0KICAgICAgICAuLi4Kcm91dGU6CiAgZ3JvdXBfYnk6IFsnam9iJ10KICBncm91cF93YWl0OiAzMHMKICBncm91cF9pbnRlcnZhbDogNW0KICByZXBlYXRfaW50ZXJ2YWw6IDEyaAogIHJlY2VpdmVyOiAnc2xhY2snCg==
+$ CONFIG_MANIFEST=$(oc create secret --dry-run=client generic alertmanager-default --from-file=alertmanager.yaml -o json)
+$ oc patch stf default --type=merge -p '{"spec":{"alertmanagerConfigManifest":'"$CONFIG_MANIFEST"'}}'
 ----
-
 . Verify that the configuration has been applied to the secret:
 +
+[NOTE]
+There will be a short delay as the operators update each object
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_creating-an-alert-route-with-templating-in-alertmanager.adoc:73]:
```suggestion
There will be a short delay as the operators update each object.
```

File: doc-Service-Telemetry-Framework/modules/proc_retrieving-and-setting-grafana-login-credentials.adoc [hunk 10]
```diff
@@ -2,7 +2,9 @@
 = Retrieving and setting Grafana login credentials
 
 [role="_abstract"]
-{Project} ({ProjectShort}) sets default login credentials when Grafana is enabled. You can override the credentials in the `ServiceTelemetry` object.
+When Grafana is enabled, you can login using openshift authentication, or the default username and password set by the Grafana Operator.
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_retrieving-and-setting-grafana-login-credentials.adoc:5]:
```suggestion
When Grafana is enabled, you can login using OpenShift authentication, or the default username and password set by the Grafana Operator.
```

File: doc-Service-Telemetry-Framework/modules/proc_retrieving-and-setting-grafana-login-credentials.adoc (hunk 6 above)
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_retrieving-and-setting-grafana-login-credentials.adoc:31]:
```suggestion
. Wait for the grafana pod to restart with the new credentials in place:
```

File: doc-Service-Telemetry-Framework/modules/proc_editing-the-metrics-retention-time-period-in-service-telemetry-framework.adoc [hunk 11]
```diff
@@ -48,6 +48,19 @@ spec:
 ----
 
 . Save your changes and close the object.
+. Wait for prometheus to restart with the new settings.
++
+[source,bash]
+----
+$ oc get po -l app.kubernetes.io/name=prometheus -w
+----
+. Verify the new retention setting by checking the command line arguments used in the pod.
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_editing-the-metrics-retention-time-period-in-service-telemetry-framework.adoc:57]:
```suggestion
. Verify the new retention setting by checking the command line arguments used in the pod:
```

File: doc-Service-Telemetry-Framework/modules/proc_editing-the-metrics-retention-time-period-in-service-telemetry-framework.adoc [hunk 12]
```diff
@@ -48,6 +48,19 @@ spec:
 ----
 
 . Save your changes and close the object.
+. Wait for prometheus to restart with the new settings.
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_editing-the-metrics-retention-time-period-in-service-telemetry-framework.adoc:51]:
```suggestion
. Wait for prometheus to restart with the new settings:
```

**Review by leifmadsen — APPROVED** [review]
Some minor syntax suggestions, but nothing significant, so I'm good with this to merge. Thanks!

**Pull request merged**